src/snprintf.c

```diff
@@ -56,12 +56,13 @@ rcutils_vsnprintf(char * buffer, size_t buffer_size, const char * format, va_lis
 #ifndef _WIN32
   ret = vsnprintf(buffer, buffer_size, format, args);
 #else
-  errno_t errno_ret = _vsnprintf_s(buffer, buffer_size, _TRUNCATE, format, args);
-  if (-1 == errno_ret && 0 == errno) {
+  // errno isn't explicitly set to 0 when truncation occurs.
+  errno = 0;
+  ret = _vsnprintf_s(buffer, buffer_size, _TRUNCATE, format, args);
```

Comment:
Why the change to not use `errno_t`?

Reply:
MSDN says that the return value is an `int`: https://docs.microsoft.com/en-us/cpp/c-runtime-library/reference/vsnprintf-s-vsnprintf-s-vsnprintf-s-l-vsnwprintf-s-vsnwprintf-s-l?view=vs-2019

Here's the definition in the Windows 10 SDK 10.0.17763.0:
```c++
    _Success_(return >= 0)
    _Check_return_opt_
    _CRT_STDIO_INLINE int __CRTDECL _vsnprintf_s(
        _Out_writes_opt_(_BufferCount) _Always_(_Post_z_) char*       const _Buffer,
        _In_                                              size_t      const _BufferCount,
        _In_                                              size_t      const _MaxCount,
        _In_z_ _Printf_format_string_                     char const* const _Format,
                                                          va_list           _ArgList
        )
```

Reply:
Unless there's some other reason to use `errno_t` that I'm not aware of?

Reply:
Hmm, I dunno, I assumed it actually returned `errno_t` and you were just simplifying it because it was an alias or something, but since it is documented as int, that is fine.